src/Raven.Server/ServerWide/ClusterStateMachine.License.cs

```diff
@@ -53,343 +55,438 @@ private void AssertLicenseLimits(string type, ServerStore serverStore, DatabaseR
+                if (AssertClientConfiguration(serverStore.LicenseManager.LicenseStatus, context) == false)
+                    throw new LicenseLimitException(LimitType.ClientConfiguration, "Your license doesn't support adding the client configuration.");
+                break;
 
-                    throw new LicenseLimitException(LimitType.CustomAnalyzers, $"The maximum number of analyzers per database cannot exceed the limit of: {maxAnalyzersPerDatabase}");
-                }
+            case nameof(PutDatabaseStudioConfigurationCommand):
+                AssertDatabaseStudioConfiguration(databaseRecord, serverStore.LicenseManager.LicenseStatus, context);
+                break;
 
-                var maxAnalyzersPerCluster = serverStore.LicenseManager.LicenseStatus.MaxNumberOfCustomAnalyzersPerCluster;
-                if (maxAnalyzersPerCluster != null && maxAnalyzersPerCluster >= 0)
-                {
-                    var totalAnalyzersCount = GetTotal(DatabaseRecordElementType.Analyzers, databaseRecord.DatabaseName) + databaseRecord.Analyzers.Count;
-                    if (totalAnalyzersCount <= maxAnalyzersPerCluster)
-                        return;
+            case nameof(PutServerWideStudioConfigurationCommand):
+                if (AssertServerWideStudioConfiguration(serverStore.LicenseManager.LicenseStatus, context) == false)
+                    throw new LicenseLimitException(LimitType.StudioConfiguration, "Your license doesn't support adding the studio configuration.");
+                break;
 
-                    if (CanAssertLicenseLimits(context, minBuildVersion: MinBuildVersion60000) == false)
-                        return;
+            case nameof(AddQueueSinkCommand):
+            case nameof(UpdateQueueSinkCommand):
+                if (AssertQueueSink(serverStore.LicenseManager.LicenseStatus, context) == false)
+                    throw new LicenseLimitException(LimitType.QueueSink, "Your license doesn't support using the queue sink feature.");
+                break;
 
-                    throw new LicenseLimitException(LimitType.CustomAnalyzers, $"The maximum number of analyzers per cluster cannot exceed the limit of: {maxAnalyzersPerCluster}");
-                }
+            case nameof(EditDataArchivalCommand):
+                if (AssertDataArchival(serverStore.LicenseManager.LicenseStatus, context) == false )
+                    throw new LicenseLimitException(LimitType.DataArchival, "Your license doesn't support using the data archival feature.");
                 break;
+        }
+    }
 
-            case nameof(UpdatePeriodicBackupCommand):
-                if (serverStore.LicenseManager.LicenseStatus.HasPeriodicBackup)
-                    return;
+    private void AssertLicense(ClusterOperationContext context, string type, BlittableJsonReaderObject bjro, ServerStore serverStore)
+    {
+        var command = (PutLicenseCommand)CommandBase.CreateFrom(bjro);
```

Comment:
We can only start this from the next version (60001?)

Reply:
done